osmoutils/binary_search.go

```diff
@@ -132,6 +150,13 @@ func BinarySearch(f func(input sdk.Int) (sdk.Int, error),
 	return sdk.Int{}, errors.New("hit maximum iterations, did not converge fast enough")
 }
 
+// SdkDec
+type SdkDec[D any] interface {
```

Comment:
What is this used for?

Reply:
Not yet used, was for trying to make this into an interfaec